tests/functional/integration-tests.js

```diff
@@ -0,0 +1,91 @@
+/*globals casper */
+
+// Test the various integration pages.
+//
+// We limit ourselves to one test per integration page. This may seem low,
+// but keep in mind that functional tests are not supposed to be as exhaustive
+// as unit tests. Instead, their role is to look at the system as a whole.
+// Nevertheless, we test a variety of features of FormatJS and multiple locales
+// to try and get the highest possible coverage and confidence that the site is
+// working properly.
+//
+// All the tests basically do the same thing. Load a page, modify the value
+// of the locale selection combo box, and check the output of the corresponding
+// example with a static value. Because these tests are all the same, they are
+// parameterized below.
+
+'use strict';
+
+var testData = {
+    'Test React integration example': {
+        comment: 'Test React integration - Relative time formatting example using ja-JP',
+        type: 'react',
+        locale: 'ja-JP',
+        id: 'ex-react-relative',
+        output_selector: 'li:nth-child(2)',
```

Comment:
It's not pretty, but I think this is the right way to write these tests. A functional test should not know the page is using a particular library. It should test expected results in the actual DOM.